lms/djangoapps/certificates/api.py

```diff
@@ -91,6 +91,42 @@ def get_certificates_for_user(username):
     ]
 
 
+def get_certificate_for_user(username, course_key):
+    """
+    Retrieve certificate information for a particular user for a specific course.
+
+    Arguments:
+        username (unicode): The identifier of the user.
+        course_key (CourseKey): A Course Key.
+    Returns: dict
+    """
+    try:
+        cert = GeneratedCertificate.eligible_certificates.get(
+            user__username=username,
+            course_id=course_key
+        )
+    except GeneratedCertificate.DoesNotExist:
+        return None
+    return {
+        "username": username,
+        "course_key": unicode(cert.course_id),
+        "type": cert.mode,
+        "status": cert.status,
+        "grade": cert.grade,
+        "created": cert.created_date,
+        "modified": cert.modified_date,
+
+        # NOTE: the download URL is not currently being set for webview certificates.
+        # In the future, we can update this to construct a URL to the webview certificate
+        # for courses that have this feature enabled.
+        "download_url": (
+            cert.download_url or get_certificate_url(cert.user.id, cert.course_id)
+            if cert.status == CertificateStatuses.downloadable
+            else None
+        ),
+    }
```

Comment:
@giocalitri good catch about the problem with the other docstring.  However, since the only use of this new function is in the new HTTP view, it seems you could easily return the (non-unicodified) course_key from this function here, like the other version does, and instead do `unicode()` on it where you generate the HTTP response ([here](https://github.com/edx/edx-platform/pull/12055/files#diff-df8c242937b095f68c277ce9d5807cdcR106)).

That the earlier function is not covered by any tests is disappointing.  It would certainly be a great help if you could add a bit of coverage considering the strong similarity with what you've implemented here.  However, in the worst case, the two places I see that function being used are in methods that themselves have tests, so we should be covered in case of any regression.